File: LeetCode/Mock Interviews/Search and Sort/search_insert_position_hari.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if(target<nums[0])
        {
            return 0;
        }
        if(target>nums[nums.size()-1])
        {
            return nums.size();
        }
        int l = 0;
        int r=nums.size();
        int mid;
        while(l<=r)
        {
            mid=(l+r)/2;
            if(nums[mid]==target)
            {
                return mid;
            }
            if(nums[mid]>target)
            {
                r=mid-1;
            }
            else if(nums[mid]<target)
            {
                l=mid+1;
            }
        }
        return l;
    }
};

File: LeetCode/Algorithms_1/search_insert_position_hari.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        if(target<nums[0])
            return 0;
        int l=0;
        int r=nums.length-1;
        if(target>nums[r]){
            return r+1;
        }
        int mid=0;
        while(l<=r){
            mid=l+(r-l)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if(nums[mid]<target){
                l=mid+1;
            }
            else
            {
                r=mid-1;
            }
        }
        return l;
    }
}
